### PR TITLE
Doc yinleon rebase

### PR DESCRIPTION
**Pull request by tacaswell**
Rebase of #8209

attn @dstansby @choldgraf

### CODE REVIEW

File: examples/pylab_examples/geo_demo.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-========
-Geo Demo
-========
+======================
+Geographic Projections
+======================
 
+This shows 4 possible projections using subplot.
+Matplotlib also supports
+<a href='http://matplotlib.org/basemap/'>Basemaps Toolkit</a> and
```
Comment (anntzer) [inline examples/pylab_examples/geo_demo.py]: use proper rst markup instead

File: examples/pylab_examples/bar_stacked.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
 """
-===========
-Bar Stacked
-===========
+=================
+Stacked Bar Graph
+=================
+
+This is an example of creating a stacked bar plot with error bars
+using `plt.bar`.  Note the parameters *yerr* used for error bars, and
```
Comment (anntzer) [inline examples/pylab_examples/bar_stacked.py]: I *guess* sphinx will fail to resolve what `plt.bar` is but I may be wrong.

File: examples/mplot3d/mixed_subplots.py [hunk 3]
```diff
@@ -1,9 +1,9 @@
 """
-==================
-2D and 3D subplots
-==================
+============================================
+Demonstrate the mixing of 2d and 3d subplots
```
Comment (anntzer) [inline examples/mplot3d/mixed_subplots.py]: 2D, 3D (see just below)

File: examples/mplot3d/lorenz_attractor.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
 '''
-====================
-The Lorenz Attractor
-====================
+================
+Lorenz Attractor
+================
+
+This is an example of plotting Edward Lorenz's 1963 "Deterministic
+Nonperiodic Flow" in a 3-dimensional space using mplot3d.
```
Comment (anntzer) [inline examples/mplot3d/lorenz_attractor.py]: link below could use proper rst markup too

File: examples/event_handling/resample.py [hunk 5]
```diff
@@ -1,9 +1,15 @@
 """
-========
-Resample
-========
+===============
+Resampling Data
+===============
+
+
```
Comment (anntzer) [inline examples/event_handling/resample.py]: remove blank line

File: examples/event_handling/resample.py [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
+Downsampling lowers the sample rate or sample size of a signal. In
+this tutorial, the signal is downsampled when the plot is adjusted
+through dragging and zooming.
```
Comment (anntzer) [inline examples/event_handling/resample.py]: ... except that the example is completely broken (try it).  It should either be fixed or removed.

Comment (tacaswell) [inline examples/event_handling/resample.py]: It does what it claims, but in a weird way.  It always down-samples to every 5th point but also clips the data.

**Comment by anntzer** [comment]
(feel free to dismiss the review when you consider the issues addressed)

File: examples/event_handling/resample.py (hunk 6 above)
Comment (tacaswell) [inline examples/event_handling/resample.py]: re worked this a bit.

It works on zoom, but not pan which seems like a bug Mpl, not in the example.

File: examples/event_handling/poly_editor.py [hunk 7]
```diff
@@ -4,8 +4,7 @@
 ===========
 
 This is an example to show how to build cross-GUI applications using
-matplotlib event handling to interact with objects on the canvas
-
+matplotlib event handling to interact with objects on the canvas.
```
Comment (dstansby) [inline examples/event_handling/poly_editor.py]: matplotlib --> Matplotlib

File: examples/mplot3d/mixed_subplots.py [hunk 8]
```diff
@@ -1,11 +1,10 @@
 """
-==================
-2D and 3D subplots
-==================
+============================================
+Demonstrate the mixing of 2D and 3D subplots
```
Comment (dstansby) [inline examples/mplot3d/mixed_subplots.py]: I think the original title is more succinct and clear, but happy for it to change

Comment (dstansby) [inline examples/mplot3d/mixed_subplots.py]: Perhaps just "Mixing 2D and 3D subplots"?

File: examples/pylab_examples/bar_stacked.py [hunk 9]
```diff
@@ -1,10 +1,16 @@
...
+
+This is an example of creating a stacked bar plot with error bars
+using `~matplotlib.pyplot.bar`.  Note the parameters *yerr* used for
+error bars, and *bottom* to stack the women's bars on top of the men's
+bars.
 
-A stacked bar plot with errorbars.
 """
+
+# a stacked bar plot with errorbars
```
Comment (dstansby) [inline examples/pylab_examples/bar_stacked.py]: This comment seems redundant now there's a proper description at the top.

File: examples/pylab_examples/geo_demo.py [hunk 10]
```diff
@@ -1,9 +1,14 @@
 """
-========
-Geo Demo
-========
+======================
+Geographic Projections
+======================
+
+This shows 4 possible projections using subplot.  Matplotlib also
+supports `Basemaps Toolkit <http://matplotlib.org/basemap>`_ and
```
Comment (dstansby) [inline examples/pylab_examples/geo_demo.py]: https

**Comment by tacaswell** [comment]
The scitools https gives we errors so leaving as http.

**Pull request merged**